caffe2/core/nomnigraph/Converters/Caffe2.cc

```diff
@@ -0,0 +1,327 @@
+#include "nomnigraph/Converters/Caffe2.h"
+#include "nomnigraph/Graph/Algorithms.h"
+
+#include "nomnigraph/Support/Casting.h"
+#include "nomnigraph/Support/Pointer.h"
+
+std::map<std::string, caffe2::Argument>
+getArgumentsFromOperator(caffe2::OperatorDef op) {
+  std::map<std::string, caffe2::Argument> argMap;
+  for (auto arg : op.arg()) {
+    argMap[arg.name()] = arg;
+  }
+  return argMap;
+}
+
+namespace nom {
+namespace converters {
+
+std::unique_ptr<repr::NeuralNetOperator>
+convertOperator(caffe2::OperatorDef op) {
+  auto argMap = getArgumentsFromOperator(op);
+
+  if (op.type() == "Conv") {
+    // There are literally three ways to define shapes in Conv in Caffe2
+    std::vector<int> kernelShape;
+    if (argMap.count("kernel")) {
+      int kernel = static_cast<int>(argMap["kernel"].i());
+      kernelShape = {kernel, kernel};
+    } else if (argMap.count("kernels")) {
+      for (auto i : argMap["kernels"].ints()) {
+        kernelShape.push_back(static_cast<int>(i));
+      }
+    } else if (argMap.count("kernel_h") && argMap.count("kernel_w")) {
+      int kernelH = static_cast<int>(argMap["kernel_h"].i());
+      int kernelW = static_cast<int>(argMap["kernel_w"].i());
+      kernelShape = {kernelH, kernelW};
+    } else {
+      assert(0);
```

Comment:
Please use `CAFFE_THROW`. `assert` is a no-op in Release mode, which is now default.

Reply:
this choice was inspired by the LLVM codebase, which doesn't use exceptions to reduce codesize and potentially speed up the program